From 1708d277a9f0b833cefe072a771589cdf7ae8367 Mon Sep 17 00:00:00 2001
From: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Date: Fri, 3 May 2013 17:11:20 +0200
Subject: AudioPolicy: fix stopInput

BZ: 104921
[CLEAN]

Given the convention for audio device identification since JB MR1 code
drop, the bit 31 of the bit field is reserved and when set indicates
an input device. So 0x0 (AUDIO_DEVICE_NONE) is identified as an output
audio device. Thus the stopInput function does not stop the audio input
stream.

In order to reroute the audio input stream as expected, this patch
modifies the stopInput function in order to use the
AUDIO_DEVICE_BIT_IN (0x80000000) value.

Change-Id: Ib87df5340ae538f2be2150a8d599e457e04976fd
Orig-Change-Id: I04fc1bdb1b778c1ff5a9ab6edc6893a436fc5ea6
Category: aosp improvement
Domain: AudioComms-AudioPolicy
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 788b93e..d009706 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -1079,7 +1079,8 @@ status_t AudioPolicyManagerBase::stopInput(audio_io_handle_t input)
         }
 
         AudioParameter param = AudioParameter();
-        param.addInt(String8(AudioParameter::keyRouting), 0);
+        // AUDIO_DEVICE_BIT_IN value allows to stop any audio input stream
+        param.addInt(String8(AudioParameter::keyRouting), AUDIO_DEVICE_BIT_IN);
         mpClientInterface->setParameters(input, param.toString());
         inputDesc->mRefCount = 0;
         return NO_ERROR;
-- 
1.7.4.1

